query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/select/SelectOrderingTest.java

```diff
@@ -46,6 +47,103 @@ public void should_generate_ordering_clauses() {
+                .orderBy("c1", ASC)
+                .orderBy("c2", DESC))
+        .hasCql("SELECT * FROM foo WHERE k=1 OR l=2 ORDER BY c1 ASC,c2 DESC");
+  }
+
+  @Test
+  public void should_support_sub_condition() {
+    assertThat(
+            selectFrom("foo")
+                .all()
+                .where(Relation.column("k").isEqualTo(literal(1)))
+                .and()
+                .where(
+                    subCondition()
+                        .where(Relation.column("l").isEqualTo(literal(2)))
+                        .or()
+                        .where(Relation.column("m").isEqualTo(literal(3))))
+                .orderBy("c1", ASC)
+                .orderBy("c2", DESC))
+        .hasCql("SELECT * FROM foo WHERE k=1 AND (l=2 OR m=3) ORDER BY c1 ASC,c2 DESC");
+    assertThat(
+            selectFrom("foo")
+                .all()
+                .where(
+                    subCondition()
+                        .where(Relation.column("l").isEqualTo(literal(2)))
+                        .where(Relation.column("m").isEqualTo(literal(3))))
+                .and()
+                .where(Relation.column("k").isEqualTo(literal(1)))
+                .orderBy("c1", ASC))
+        .hasCql("SELECT * FROM foo WHERE (l=2 AND m=3) AND k=1 ORDER BY c1 ASC");
+    assertThat(
+            selectFrom("foo")
+                .all()
+                .where(
+                    subCondition()
+                        .where(Relation.column("l").isEqualTo(literal(2)))
+                        .where(Relation.column("m").isEqualTo(literal(3)))
+                        .where(Relation.column("x").isEqualTo(literal(4))))
+                .orderBy("c1", ASC))
+        .hasCql("SELECT * FROM foo WHERE (l=2 AND m=3 AND x=4) ORDER BY c1 ASC");
+    assertThat(
+            selectFrom("foo")
+                .all()
+                .where(Relation.column("k").isEqualTo(literal(1)))
+                .where(
+                    subCondition()
+                        .where(Relation.column("l").isEqualTo(literal(2)))
+                        .where(Relation.column("m").isEqualTo(literal(3)))
+                        .or()
+                        .where(Relation.column("x").isEqualTo(literal(4)))))
+        .hasCql("SELECT * FROM foo WHERE k=1 AND (l=2 AND m=3 OR x=4)");
+  }
+
+  @Test
+  public void should_use_conjunction_as_default_logical_operator() {
+    assertThat(
+            selectFrom("foo")
+                .all()
+                .where(Relation.column("k").isEqualTo(literal(1)))
+                .where(
+                    subCondition()
+                        .where(Relation.column("l").isEqualTo(literal(2)))
+                        .or()
+                        .where(Relation.column("m").isEqualTo(literal(3))))
+                .orderBy("c1", ASC)
+                .orderBy("c2", DESC))
+        .hasCql("SELECT * FROM foo WHERE k=1 AND (l=2 OR m=3) ORDER BY c1 ASC,c2 DESC");
+  }
```

Comment:
Is it possible we can add some more complex logical query in tests.

Reply:
Replaced last test to be more complex. Above tests also verify backward-compatibility of current default `AND` operator.